src/DefaultInjectablesFactory.php

```diff
@@ -135,4 +137,20 @@ function(Context $context, $paramName) {
 			}
 		);
 	}
+
+	public function createFileUploadParam() {
+		return new InjectableProperty(
+			'Fliglio\Web\FileUpload',
+			function(Context $context, $paramName) {
+				$files = $context->getRequest()->getFiles();
+
+				if (!isset($files[$paramName])) {
```

Comment:
Not entirely obvious how the method signature/argument $paramName correlates to an uploaded file input?
e.g. 
```
Resource::uploadedFile(FileUpload $anyRandoName)
```

must match:
```
<input type='file' name='anyRandoName' />
```

???

Reply:
thats exactly right, it follows the convention used by get params & path params. 

`$_FILES[$paramName]` targets a specific file upload

should I add something to make that more explicit?

Reply:
...and you target it with the name of your injected param:
```
public function myResourceMethod(FileUpload $anyRandoName) { ... }
```